pkg/ccip/chainaccessor/ton_accessor.go

```diff
@@ -540,39 +579,57 @@ func (a *TONAccessor) ExecutedMessages(
 	executed := make(map[ccipocr3.ChainSelector][]ccipocr3.SeqNum)
 
 	// Query for ExecutionStateChanged events for all chains/ranges
-	// TODO(@jadepark-dev): revisit when we have a persistent log DB implemented(should be OR condition with SQL query)
+	// TODO(@jadepark-dev): Note: Currently iterating per chain/range - optimize with OR conditions
```

Comment:
might want to run those query concurrently for each chain, if we are querying many chains. This can probably wait until we moving those query into its own layer suggested [above](https://github.com/smartcontractkit/chainlink-ton/pull/219/files#r2501271553), as current logic is pretty verbose already.